tests/datatypes/test_frozenset.py

```diff
@@ -15,6 +15,24 @@ def test_len(self):
         print(len(frozenset([1, 2])))
         """)
 
+    def test_union(self):
+        self.assertCodeExecution("""
+        x = frozenset([2,4,5,6])
+        y = frozetset([3,4,7,9])
+        z = 5
+        print(x.union(y))
+        print(x.union(z))
```

Comment:
This is a good test, and it probably passes, but it's not the right way to test it in Batavia.

Batavia tests are based on comparing output of two program runs; and, to do that, it's necessary for both programs to terminate successfully, not raise an error.

If you're testing behavior that you know will (and should) raise an error, your test code should catch the exception and print a "raised an error" message; that way, the comparison has something to match up.